src/components/common/countdown-button/countdown-button.tsx

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import React, { useMemo, useState } from 'react'
+import type { ButtonProps } from 'react-bootstrap'
+import { Button } from 'react-bootstrap'
+import { useInterval } from 'react-use'
+
+export interface CountdownButtonProps extends ButtonProps {
+  countdownStartSeconds: number
+}
+
+/**
+ * Button that starts a countdown on render and is only clickable after the countdown has finished.
+ */
+export const CountdownButton: React.FC<CountdownButtonProps> = ({ countdownStartSeconds, children, ...props }) => {
+  const [secondsRemaining, setSecondsRemaining] = useState(countdownStartSeconds)
+
+  useInterval(
+    () => setSecondsRemaining((previous) => previous - 1),
+    useMemo(() => (secondsRemaining <= 0 ? null : 1000), [secondsRemaining])
```

Comment:
You can remove the memo here. The value is easy to calculate and the type is a primitive.